### ISSUE TITLE
Mosh clipboard issue

### ISSUE THREAD

**Issue by jiaxincao** [mobile-shell/mosh#1090]
https://github.com/mobile-shell/mosh/blob/0cc492dbae2f6aaef9a54dc2a8ba3222868b150f/src/terminal/terminaldisplay.cc#L114

It seems like that mosh is trying to create a clipboard caching to avoid OSC 52 clipboard delivery if the text is the same with the last time. However, this introduces a problem. Think of the following scenario, I'm using emacs in a remote server from my laptop via mosh, and trying to copy texts with OSC 52. I took the following steps.
1. My remote emacs sends a OSC 52 to deliver abc as the clipboard text. My local laptop clipboard and the mosh clipboard are both abc for now. 
2. I copy something else (e.g., xyz) from another application on my local side, say, chrome. My local laptop clipboard is now xyz. The mosh clipboard is unchanged, which is still abc.
3. I went back to my remote emacs, and tries to send a OSC 52 to deliver abc to my local terminal again. However, this time, since the mosh clipboard still has the abc, it'll refuse to deliver OSC 52 with text abc to my terminal, which means that abc will not be copied to my local laptop clipboard.

I think the fundamental issue here is that mosh tries to do the clipboard caching, while there is no guarantee that the mosh cached clipboard is in sync with the local system clipboard. Anyone agree on this? Any workaround?

**Comment by andersk** [comment]
Yeah, this is a bug in #899 that I [pointed out before](https://github.com/mobile-shell/mosh/issues/637#issuecomment-362938406) (but thanks for opening an issue for it).

**Comment by jiaxincao** [comment]
@andersk , I think we are not talking about the same issue. I see the issue reported even with your fix. Mosh refuses to transfer any OSC 52 code to the terminal if it was the same one with the last time.

**Comment by andersk** [comment]
We’re talking about the same issue: see item 2 of my [comment](https://github.com/mobile-shell/mosh/issues/637#issuecomment-362938406) that I linked. I didn’t say it was fixed, much less by me.

**Comment by jiaxincao** [comment]
Ah, you are right. So it seems like this is an issue that is still opening, correct?

Do you know why a clipboard cache was introduced in mosh? Why can we just sent the OSC 52 to the terminal only when the remote shell (or terminal applications) decides to do it?
